src/rdpc/fetchDonorIdsByAnalysis.ts

```diff
@@ -31,14 +31,17 @@ type QueryVariable = {
 const fetchDonorIdsByAnalysis = async ({
```

Comment:
even though technically the JWT has been checked for expiry, there's a slim but non-zero chance that it could expire in the middle of a stream. I think it's best to just handle the jwt retrieval in the fetchers. 

It's also possible that the JWT expires mid-flight on its way to the RDPC, so we still need to catch the `403` and retry. If we use the design I left above, the retry should automatically come with a new JWT

Reply:
the problem is rdpc doesn't return a specific code for expired token, `Permission Denied` is being returned now if a token has invalid scope, so if we retry on permission denied we are only sending another invalid token. Is it better if rdpc can return a code for expired token so aggregator can fetch a new token?

Reply:
if you use the `createEgoJwtManager` above, you'll be guaranteed to have the updated jwt, cause the `getLatestJwt`  method basically takes care of that